lighthouse/src/main.rs

```diff
@@ -50,6 +52,18 @@ fn bls_library_name() -> &'static str {
     }
 }
 
+#[inline(always)]
+#[allow(unreachable_code)]
```

Comment:
```suggestion
```

If you write the body of the function with mutually exclusive `cfg`s so that there's only 1 return statement, then this won't be necessary. I think removing the `cfg(feature = "portable")` will get most of the way there.